tests/unittest_builder.py

```diff
@@ -768,7 +768,10 @@ def test_module_base_props(self) -> None:
         """test base properties and method of an astroid module"""
         module = self.module
         self.assertEqual(module.name, "data.module")
-        self.assertEqual(module.doc, "test module for astroid\n")
+        with pytest.warns(DeprecationWarning) as records:
+            self.assertEqual(module.doc, "test module for astroid\n")
+            assert len(records) == 1
+        self.assertEqual(module.doc_node.value, "test module for astroid\n")
```

Comment:
Should we make these test type safe by adding `isinstance(module.doc_node, nodes.Const)`?

Reply:
Hm, we could but aren't we testing this in other places as well? Feels a bit like redundant double testing, but I'll happily change!

Reply:
I'm split on this one. It wouldn't be strictly necessary, but maybe a bit better for type checking.

Reply:
I'll change. Give me a coupe of minutes!

Reply:
See latest commit.